// the-link/shim/esm/TheLinkStyle.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import c from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { TheStyle } from 'the-style';
/** Style for TheLink */

var TheLinkStyle = function TheLinkStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return React.createElement(TheStyle, _extends({
    id: id
  }, {
    className: c('the-link-style', className),
    styles: TheLinkStyle.data(options)
  }));
};

TheLinkStyle.displayName = 'TheLinkStyle';
TheLinkStyle.propTypes = {
  /** Style options */
  options: PropTypes.object
};
TheLinkStyle.defaultProps = {
  options: {}
};

TheLinkStyle.data = function (options) {
  var ThemeValues = TheStyle.ThemeValues;
  var _options$dominantColo = options.dominantColor,
      dominantColor = _options$dominantColo === void 0 ? ThemeValues.dominantColor : _options$dominantColo;
  return {
    '.the-link': {
      color: dominantColor,
      display: 'inline-block',
      padding: '0px 4px'
    },
    '.the-link-active': {
      textDecoration: 'none'
    }
  };
};

export default TheLinkStyle;
