// the-link/shim/esm/TheLink.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import c from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';
import { eventHandlersFor, htmlAttributesFor } from 'the-component-util';
import { TheIcon } from 'the-icon';
import { parse as parseUrl } from 'url';
import TheLinkStyle from './TheLinkStyle';
/**
 * Link of the-components
 */

var TheLink =
/*#__PURE__*/
function (_React$Component) {
  _inherits(TheLink, _React$Component);

  function TheLink() {
    _classCallCheck(this, TheLink);

    return _possibleConstructorReturn(this, _getPrototypeOf(TheLink).apply(this, arguments));
  }

  _createClass(TheLink, [{
    key: "render",
    value: function render() {
      var props = this.props;
      var activeClassName = props.activeClassName,
          activeStyle = props.activeStyle,
          children = props.children,
          className = props.className,
          color = props.color,
          exact = props.exact,
          icon = props.icon,
          replace = props.replace,
          strict = props.strict,
          _props$style = props.style,
          style = _props$style === void 0 ? {} : _props$style,
          to = props.to;

      if (color && style) {
        style.color = color;
      }

      return React.createElement(TheLink.Link, _extends({}, htmlAttributesFor(props, {
        except: ['className', 'style']
      }), eventHandlersFor(props, {
        except: []
      }), {
        activeClassName: c('the-link-active', activeClassName),
        className: c('the-link', className)
      }, {
        activeStyle: activeStyle,
        exact: exact,
        replace: replace,
        strict: strict,
        style: style,
        to: to
      }), icon && React.createElement(TheIcon, {
        className: c('the-link-icon', icon)
      }), children);
    }
  }], [{
    key: "Link",
    value: function Link(props) {
      var children = props.children,
          className = props.className,
          to = props.to;

      var _ref = to && parseUrl(to) || {},
          protocol = _ref.protocol;

      if (protocol) {
        // External link
        return React.createElement("a", _extends({}, htmlAttributesFor(props, {
          except: ['className', 'href']
        }), eventHandlersFor(props, {
          except: []
        }), {
          className: c('the-link-external', className),
          href: to
        }), children);
      }

      return React.createElement(NavLink, props, children);
    }
  }]);

  return TheLink;
}(React.Component);

TheLink.Style = TheLinkStyle;
TheLink.propTypes = {
  /** Class name for active state */
  activeClassName: PropTypes.string,

  /** Style for active state */
  activeStyle: PropTypes.object,

  /** Color theme */
  color: PropTypes.string,

  /** Exact path */
  exact: PropTypes.bool,

  /** Icon class name */
  icon: PropTypes.string,

  /** String path */
  strict: PropTypes.bool,

  /** Link to */
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired
};
TheLink.defaultProps = {
  activeClassName: null,
  activeStyle: {},
  color: null,
  exact: false,
  icon: null,
  replace: false,
  role: 'link',
  strict: false,
  to: null
};
TheLink.displayName = 'TheLink';
export default TheLink;
