// the-container/shim/esm/TheContainerStyle.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import c from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { TheStyle } from 'the-style';
/** Style for TheContainer */

var TheContainerStyle = function TheContainerStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return React.createElement(TheStyle, _extends({
    id: id
  }, {
    className: c('the-container-style', className),
    styles: TheContainerStyle.data(options)
  }));
};

TheContainerStyle.displayName = 'TheContainerStyle';
TheContainerStyle.propTypes = {
  /** Style options */
  options: PropTypes.object
};
TheContainerStyle.defaultProps = {
  options: {}
};

TheContainerStyle.data = function (options) {
  var ThemeValues = TheStyle.ThemeValues;
  var _options$containerWid = options.containerWidth,
      containerWidth = _options$containerWid === void 0 ? ThemeValues.containerWidth : _options$containerWid;
  return {
    '.the-container': {
      margin: '0 auto',
      maxWidth: containerWidth
    }
  };
};

export default TheContainerStyle;
