// the-footer/shim/esm/TheFooter.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { htmlAttributesFor } from 'the-component-util';
import { TheContainer } from 'the-container';
import { TheLink } from 'the-link';
import TheFooterStyle from './TheFooterStyle';
/**
 * Footer of the-components
 */

var TheFooter =
/*#__PURE__*/
function (_React$PureComponent) {
  _inherits(TheFooter, _React$PureComponent);

  function TheFooter() {
    _classCallCheck(this, TheFooter);

    return _possibleConstructorReturn(this, _getPrototypeOf(TheFooter).apply(this, arguments));
  }

  _createClass(TheFooter, [{
    key: "render",
    value: function render() {
      var props = this.props;
      var children = props.children,
          className = props.className;
      return React.createElement("footer", _extends({}, htmlAttributesFor(props, {
        except: ['className']
      }), {
        className: classnames('the-footer', className)
      }), React.createElement(TheContainer, {
        className: "the-footer-inner"
      }, children));
    }
  }], [{
    key: "CopyRight",
    value: function CopyRight(_ref) {
      var children = _ref.children,
          className = _ref.className,
          holder = _ref.holder,
          year = _ref.year;
      return React.createElement("div", {
        className: classnames('the-footer-copyright', className)
      }, holder && "\xA9 ".concat(year || '', " ").concat(holder), children);
    }
  }, {
    key: "Link",
    value: function Link(_ref2) {
      var children = _ref2.children,
          className = _ref2.className,
          to = _ref2.to;
      return React.createElement(TheLink, _extends({
        className: classnames('the-footer-link', className)
      }, {
        to: to
      }), children);
    }
  }, {
    key: "Links",
    value: function Links(_ref3) {
      var children = _ref3.children,
          className = _ref3.className;
      return React.createElement("div", {
        className: classnames('the-footer-links', className)
      }, children);
    }
  }, {
    key: "Row",
    value: function Row(_ref4) {
      var children = _ref4.children,
          className = _ref4.className;
      return React.createElement("div", {
        className: classnames('the-footer-row', className)
      }, children);
    }
  }]);

  return TheFooter;
}(React.PureComponent);

TheFooter.Style = TheFooterStyle;
TheFooter.propTypes = {};
TheFooter.defaultProps = {
  role: 'contentinfo'
};
TheFooter.displayName = 'TheFooter';
export default TheFooter;
