// the-footer/shim/esm/TheFooterStyle.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { asStyleData } from 'the-component-util';
import { TheStyle } from 'the-style';
/** Style for TheFooter */

var TheFooterStyle = function TheFooterStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return React.createElement(TheStyle, _extends({
    id: id
  }, {
    className: classnames('the-footer-style', className),
    styles: TheFooterStyle.data(options)
  }));
};

TheFooterStyle.displayName = 'TheFooterStyle';
TheFooterStyle.propTypes = {
  /** Style options */
  options: PropTypes.object
};
TheFooterStyle.defaultProps = {
  options: {}
};

TheFooterStyle.data = function (options) {
  var ThemeValues = TheStyle.ThemeValues;
  var _options$lightBorderC = options.lightBorderColor,
      lightBorderColor = _options$lightBorderC === void 0 ? ThemeValues.lightBorderColor : _options$lightBorderC,
      _options$lightLinkCol = options.lightLinkColor,
      lightLinkColor = _options$lightLinkCol === void 0 ? ThemeValues.lightLinkColor : _options$lightLinkCol;
  return asStyleData('.the-footer', {
    '.the-footer-copyright': {
      display: 'inline-block',
      padding: '0 8px'
    },
    '.the-footer-inner': {
      alignItems: 'center',
      borderTop: "1px solid ".concat(lightBorderColor),
      display: 'flex',
      flexWrap: 'wrap',
      fontSize: 'smaller',
      justifyContent: 'space-between',
      padding: '32px 0'
    },
    '.the-footer-link': {
      color: lightLinkColor,
      padding: '0 8px',
      textDecoration: 'none'
    },
    '.the-footer-links': {
      display: 'flex'
    },
    '.the-footer-row': {
      alignItems: 'center',
      display: 'flex',
      flexWrap: 'wrap',
      fontSize: 'smaller',
      justifyContent: 'space-between',
      margin: '8px 0'
    },
    '': {}
  });
};

export default TheFooterStyle;
