// the-container/shim/TheContainerStyle.js
'use strict';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _theStyle = require("the-style");

/** Style for TheContainer */
var TheContainerStyle = function TheContainerStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return _react.default.createElement(_theStyle.TheStyle, (0, _extends2.default)({
    id: id
  }, {
    className: (0, _classnames.default)('the-container-style', className),
    styles: TheContainerStyle.data(options)
  }));
};

TheContainerStyle.displayName = 'TheContainerStyle';
TheContainerStyle.propTypes = {
  /** Style options */
  options: _propTypes.default.object
};
TheContainerStyle.defaultProps = {
  options: {}
};

TheContainerStyle.data = function (options) {
  var ThemeValues = _theStyle.TheStyle.ThemeValues;
  var _options$containerWid = options.containerWidth,
      containerWidth = _options$containerWid === void 0 ? ThemeValues.containerWidth : _options$containerWid;
  return {
    '.the-container': {
      margin: '0 auto',
      maxWidth: containerWidth
    }
  };
};

var _default = TheContainerStyle;
exports.default = _default;
