// the-footer/shim/TheFooterStyle.js
'use strict';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _theComponentUtil = require("the-component-util");

var _theStyle = require("the-style");

/** Style for TheFooter */
var TheFooterStyle = function TheFooterStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return _react.default.createElement(_theStyle.TheStyle, (0, _extends2.default)({
    id: id
  }, {
    className: (0, _classnames.default)('the-footer-style', className),
    styles: TheFooterStyle.data(options)
  }));
};

TheFooterStyle.displayName = 'TheFooterStyle';
TheFooterStyle.propTypes = {
  /** Style options */
  options: _propTypes.default.object
};
TheFooterStyle.defaultProps = {
  options: {}
};

TheFooterStyle.data = function (options) {
  var ThemeValues = _theStyle.TheStyle.ThemeValues;
  var _options$lightBorderC = options.lightBorderColor,
      lightBorderColor = _options$lightBorderC === void 0 ? ThemeValues.lightBorderColor : _options$lightBorderC,
      _options$lightLinkCol = options.lightLinkColor,
      lightLinkColor = _options$lightLinkCol === void 0 ? ThemeValues.lightLinkColor : _options$lightLinkCol;
  return (0, _theComponentUtil.asStyleData)('.the-footer', {
    '.the-footer-copyright': {
      display: 'inline-block',
      padding: '0 8px'
    },
    '.the-footer-inner': {
      alignItems: 'center',
      borderTop: "1px solid ".concat(lightBorderColor),
      display: 'flex',
      flexWrap: 'wrap',
      fontSize: 'smaller',
      justifyContent: 'space-between',
      padding: '32px 0'
    },
    '.the-footer-link': {
      color: lightLinkColor,
      padding: '0 8px',
      textDecoration: 'none'
    },
    '.the-footer-links': {
      display: 'flex'
    },
    '.the-footer-row': {
      alignItems: 'center',
      display: 'flex',
      flexWrap: 'wrap',
      fontSize: 'smaller',
      justifyContent: 'space-between',
      margin: '8px 0'
    },
    '': {}
  });
};

var _default = TheFooterStyle;
exports.default = _default;
