// the-footer/shim/TheFooter.js
'use strict';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _theComponentUtil = require("the-component-util");

var _theContainer = require("the-container");

var _theLink = require("the-link");

var _TheFooterStyle = _interopRequireDefault(require("./TheFooterStyle"));

/**
 * Footer of the-components
 */
var TheFooter =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inherits2.default)(TheFooter, _React$PureComponent);

  function TheFooter() {
    (0, _classCallCheck2.default)(this, TheFooter);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TheFooter).apply(this, arguments));
  }

  (0, _createClass2.default)(TheFooter, [{
    key: "render",
    value: function render() {
      var props = this.props;
      var children = props.children,
          className = props.className;
      return _react.default.createElement("footer", (0, _extends2.default)({}, (0, _theComponentUtil.htmlAttributesFor)(props, {
        except: ['className']
      }), {
        className: (0, _classnames.default)('the-footer', className)
      }), _react.default.createElement(_theContainer.TheContainer, {
        className: "the-footer-inner"
      }, children));
    }
  }], [{
    key: "CopyRight",
    value: function CopyRight(_ref) {
      var children = _ref.children,
          className = _ref.className,
          holder = _ref.holder,
          year = _ref.year;
      return _react.default.createElement("div", {
        className: (0, _classnames.default)('the-footer-copyright', className)
      }, holder && "\xA9 ".concat(year || '', " ").concat(holder), children);
    }
  }, {
    key: "Link",
    value: function Link(_ref2) {
      var children = _ref2.children,
          className = _ref2.className,
          to = _ref2.to;
      return _react.default.createElement(_theLink.TheLink, (0, _extends2.default)({
        className: (0, _classnames.default)('the-footer-link', className)
      }, {
        to: to
      }), children);
    }
  }, {
    key: "Links",
    value: function Links(_ref3) {
      var children = _ref3.children,
          className = _ref3.className;
      return _react.default.createElement("div", {
        className: (0, _classnames.default)('the-footer-links', className)
      }, children);
    }
  }, {
    key: "Row",
    value: function Row(_ref4) {
      var children = _ref4.children,
          className = _ref4.className;
      return _react.default.createElement("div", {
        className: (0, _classnames.default)('the-footer-row', className)
      }, children);
    }
  }]);
  return TheFooter;
}(_react.default.PureComponent);

TheFooter.Style = _TheFooterStyle.default;
TheFooter.propTypes = {};
TheFooter.defaultProps = {
  role: 'contentinfo'
};
TheFooter.displayName = 'TheFooter';
var _default = TheFooter;
exports.default = _default;
