// the-router/shim/esm/TheRouter.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import c from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { BrowserRouter, HashRouter, Router as AbstractRouter, StaticRouter } from 'react-router-dom';
import { htmlAttributesFor } from 'the-component-util';
/**
 * Router of the-components
 */

var TheRouter =
/*#__PURE__*/
function (_React$Component) {
  _inherits(TheRouter, _React$Component);

  function TheRouter() {
    _classCallCheck(this, TheRouter);

    return _possibleConstructorReturn(this, _getPrototypeOf(TheRouter).apply(this, arguments));
  }

  _createClass(TheRouter, [{
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this = this;

      var history = this.props.history;

      if (history) {
        var historyChanged = prevProps.history && prevProps.history !== history;

        if (historyChanged) {
          throw new Error('[TheRouter] You cannot change <TheRouter history>');
        }

        this.unlisten = history.listen(function () {
          _this.forceUpdate();
        });
      }
    }
  }, {
    key: "componentWillMount",
    value: function componentWillMount() {
      var _this2 = this;

      var _this$props = this.props,
          forceRefresh = _this$props.forceRefresh,
          history = _this$props.history;

      if (history) {
        this.unlisten = history.listen(function () {
          if (forceRefresh) {
            _this2.forceUpdate();
          }
        });
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.unlisten && this.unlisten();
    } // TODO

  }, {
    key: "getChildContext",
    value: function getChildContext() {
      var history = this.props.history;
      return {
        history: history
      };
    }
  }, {
    key: "render",
    value: function render() {
      var props = this.props;
      var basename = props.basename,
          children = props.children,
          className = props.className,
          getUserConfirmation = props.getUserConfirmation,
          history = props.history,
          keyLength = props.keyLength;
      var Router = history ? AbstractRouter : BrowserRouter;
      return React.createElement(Router, {
        basename: basename,
        getUserConfirmation: getUserConfirmation,
        history: history,
        keyLength: keyLength
      }, React.createElement("div", _extends({}, htmlAttributesFor(props, {
        except: ['className']
      }), {
        className: c('the-router', className)
      }), children));
    }
  }], [{
    key: "Hash",
    value: function Hash(_ref) {
      var basename = _ref.basename,
          children = _ref.children,
          getUserConfirmation = _ref.getUserConfirmation,
          hashType = _ref.hashType;
      return React.createElement(HashRouter, {
        basename: basename,
        getUserConfirmation: getUserConfirmation,
        hashType: hashType
      }, React.createElement("div", {
        className: "the-router-hash"
      }, children));
    }
  }, {
    key: "Static",
    value: function Static(_ref2) {
      var basename = _ref2.basename,
          children = _ref2.children,
          context = _ref2.context,
          location = _ref2.location;
      return React.createElement(StaticRouter, {
        basename: basename,
        context: context,
        location: location
      }, React.createElement("div", {
        className: "the-router-static"
      }, children));
    }
  }]);

  return TheRouter;
}(React.Component);

TheRouter.propTypes = {};
TheRouter.defaultProps = {};
TheRouter.displayName = 'TheRouter';
TheRouter.childContextTypes = {
  history: PropTypes.object
};
export default TheRouter;
